src/test/java/com/coveo/spillway/functional/SpillwayFunctionalTests.java

```diff
@@ -146,6 +148,10 @@ public void asyncPerformance() {
       lastResponse =
           asyncStorage.incrementAndGet(RESOURCE1, LIMIT1, PROPERTY1, EXPIRATION, TIMESTAMP);
     }
+    asyncStorage.shutdownStorage();
+    while (!asyncStorage.isTerminated()) {
```

Comment:
Yes, I just wasn't sure of the implementation. Is a Thread sleep of like 10ms between checks good enough?